src/OrchardCoreContrib.Linq/OrchardCoreDataContext.cs

```diff
@@ -0,0 +1,161 @@
+﻿using OrchardCore.AdminDashboard.Indexes;
+using OrchardCore.Alias.Indexes;
+using OrchardCore.AuditTrail.Indexes;
+using OrchardCore.ContentLocalization.Records;
+using OrchardCore.ContentManagement.Records;
+using OrchardCore.Deployment.Indexes;
+using OrchardCore.Layers.Indexes;
+using OrchardCore.Lists.Indexes;
+using OrchardCore.OpenId.YesSql.Indexes;
+using OrchardCore.PublishLater.Indexes;
+using OrchardCore.Taxonomies.Indexing;
+using OrchardCore.Users.Indexes;
+using OrchardCore.Workflows.Indexes;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using YesSql;
+using YesSql.Indexes;
+
+namespace OrchardCoreContrib.Linq
+{
+    /// <summary>
+    /// Represents a data context for Orchard Core index tables.
+    /// </summary>
+    public class OrchardCoreDataContext : IDisposable
```

Comment:
I'm not sure I understand the point of this. It's only to provide shortcuts to fetch full index tables? What has it to do with LINQ?

Reply:
The next step to add other useful tables

> What has it to do with LINQ?

The idea is let the users discover all the available entities / tables from a single point without looking on every part in OC. It quite similar to LinqToDb but the work that you did there is to query non OC tables, while this focus on OC related tables. It's something similar to LINQ to Umbraco where there's a context that you start from

Reply:
I see, thanks.

BTW LinqToDb is for querying anything; while usually you don't need it for index tables, you can very well use it for them:  https://github.com/Lombiq/Helpful-Libraries/blob/dev/Lombiq.HelpfulLibraries.Samples/Controllers/LinqToDbSamplesController.cs

Reply:
When I started on this I was using Lombiq.HelpfulLibraries.LinqToDb but while there's no NuGet packages I tried to do the job fast without relying on LinqToDb

The work that you did is great for many senario, but in case that you need to work with OC tables you need to have a knowledge about OC to understand what are the tables & indexes available

I may have a deep look into Lombiq.HelpfulLibraries.LinqToDb and give a feedback

Thanks

Reply:
Sure, let me know if you need help.